OpenRA.Mods.Common/ActorExts.cs

```diff
@@ -141,6 +141,10 @@ public static void NotifyBlocker(this Actor self, IEnumerable<CPos> positions)
 					return false;
 			}
 
+			// Occupied by any actor, including ally/friendly units
+			if (self.World.ActorMap.GetActorsAt(pos).Any())
```

Comment:
Ofcourse the code can become more complex but will it be worth it? In practice one or two cells occupied by enemy infantry wouldn't be a problem. Or if many cells are occupied, then the harvester is quite dead anyway.